spinalcordtoolbox/image.py

```diff
@@ -279,7 +282,22 @@ def __init__(self, param=None, hdr=None, orientation=None, absolutepath=None, di
 
         # load an image from file
         if isinstance(param, str):
-            self.loadFromPath(param, verbose)
+            try:
+                self.loadFromPath(param, mmap, verbose)
+            except OSError as e:
+                if str(e) == "[Errno 24] Too many open files":
```

Comment:
Nitpick: checking the string representation of the error may be fragile, it might be better to compare the numeric `errno` attribute:
```py3
import errno  # can be imported at the top of the module
if e.errno == errno.EMFILE:
    # Too many open files
```
This should even [work on Windows](https://docs.python.org/3/library/exceptions.html#OSError.winerror), although the `ulimit` suggestion may not be applicable there.

Reply:
Fascinating! Thank you for teaching me about the [`errno`](https://docs.python.org/3/library/errno.html) library. :slightly_smiling_face:

Reply:
Works perfectly!